tor_archivist/main.py

```diff
@@ -6,13 +6,30 @@
 from tor_core.config import config
 from tor_core.helpers import css_flair
 from tor_core.helpers import run_until_dead
-from tor_core.helpers import subreddit_from_url
 from tor_core.initialize import build_bot
 from tor_core.strings import reddit_url
 
 from tor_archivist import __version__
 
 
+def find_transcription(post):
+    """
+    Browse the top level comments of a thread, and return the first one that
+    is a transcription. Currently pretty much a copy of its counterpart in
+    ToR, which should definitely change later on.
+
+    :param post: the thread to look in.
+    :return: the matching comment, or None if it wasn't found.
+    """
+    post.comments.replace_more(limit=0)
+
+    for comment in post.comments.list():
+        if 'www.reddit.com/r/TranscribersOfReddit' in comment.body:
```

Comment:
I suppose that's one way of doing this, but what I've done for this sort of thing is this:
```python
    for comment in submission.comments:
        if footer in comment.body:
            print(f"https://reddit.com{comment.permalink}")
            transcribed.append(comment)
```

where footer is
```python
footer = """^^I'm&#32;a&#32;human&#32;volunteer&#32;content&#32;transcriber&#32;for&#32;Reddit!&#32;[If&#32;&#32;you'd&#32;like&#32;more&#32;information&#32;on&#32;what&#32;we&#32;do&#32;and&#32;why&#32;we&#32;do&#32;it,&#32;click&#32;here!](https://www.reddit.com/r/TranscribersOfReddit/wiki/index)"""
```

Reply:
This is the same footer check as is used in ToR. See also the second paragraph in the PR message - I want to merge (part of) the functionality from both bots into tor_core, to avoid duplicate code. But yeah, we only check for the link when detecting the footer.

Reply:
If it's the same, then that's fine. 

We might want that to be a more in depth check, but that's for another PR.

Reply:
No, I think it's completely fine as it is now - the only way it could fail is if someone else links to our wiki in a top-level comment on the thread, and gets more upvotes than the actual transcription. Which basically would only happen if someone deliberately tries to screw up the archive, and is just as easily done if we check for the entire header. The same counted for ToR until history validation was added.

Furthermore, until the config update is done, this would make it more difficult to change the footer (for example, by adding the word "human" :P), since we'd have to change stuff in multiple places then.

Reply:
Yeah, I just woke up so I'm going to blame it on that then. I take back what we said — it's good as it is.

However, it would be cool to move all of the checking things into tor_core (I think :P). That way we can use history validation in places like this.

Reply:
There's an open issue for moving things out of `tor_core` back into here for topics that only pertain to this bot. We'd have to look through each check to evaluate the best place for it to live.

Either way, as far as this thread goes, 👍